go.mod

```diff
@@ -59,214 +57,234 @@ require (
 	bitbucket.org/bertimus9/systemstat v0.0.0-20180207000608-0eeff89b0690
 	cloud.google.com/go v0.0.0-20160913182117-3b1ae45394a2
 	github.com/Azure/azure-sdk-for-go v21.3.0+incompatible
-	github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
+	github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78 // indirect
 	github.com/Azure/go-autorest v11.1.0+incompatible
-	github.com/GeertJohan/go.rice v0.0.0-20170420135705-c02ca9a983da
+	github.com/GeertJohan/go.rice v0.0.0-20170420135705-c02ca9a983da // indirect
 	github.com/GoogleCloudPlatform/k8s-cloud-provider v0.0.0-20181220005116-f8e995905100
 	github.com/JeffAshton/win_pdh v0.0.0-20161109143554-76bb4ee9f0ab
 	github.com/MakeNowJust/heredoc v0.0.0-20170808103936-bb23615498cd
 	github.com/Microsoft/go-winio v0.4.11
 	github.com/Microsoft/hcsshim v0.0.0-20190110205307-69ac8d3f7fc1
-	github.com/NYTimes/gziphandler v0.0.0-20170623195520-56545f4a5d46
-	github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5
-	github.com/PuerkitoBio/purell v1.0.0
-	github.com/PuerkitoBio/urlesc v0.0.0-20160726150825-5bd2802263f2
-	github.com/Rican7/retry v0.0.0-20160712041035-272ad122d6e5
+	github.com/NYTimes/gziphandler v0.0.0-20170623195520-56545f4a5d46 // indirect
+	github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5 // indirect
+	github.com/PuerkitoBio/purell v1.1.0
```

Comment:
What did you run to do this work?  Why are we blindly updating some deps?

Reply:
most of the commits have associated commands (`go mod init`, `go mod tidy`, `hack/update-vendor.sh`, etc). before this exits WIP, each commit will have a reproducible command associated/documented.

I'm actually trying to figure out why `go mod tidy` is bumping this particular dependency right now...

Reply:
turns out go aggressively rewrites your go.mod file require statements to reflect the version of the referenced that will actually be used for the build, based on `max(local require, transitive dependency require...)`

the local `replace` can still stomp and pin to a particular version (which we do), but in this case, one of our dependencies was requiring v1.1.0, so our local require statement got bumped up:

```
$ go mod graph | grep github.com/PuerkitoBio/purell
k8s.io/kubernetes github.com/PuerkitoBio/purell@v1.1.0
github.com/go-openapi/loads@v0.17.0 github.com/PuerkitoBio/purell@v1.1.0
github.com/go-openapi/spec@v0.17.2 github.com/PuerkitoBio/purell@v1.1.0
github.com/go-openapi/loads@v0.17.2 github.com/PuerkitoBio/purell@v1.1.0
github.com/go-openapi/spec@v0.17.0 github.com/PuerkitoBio/purell@v1.1.0
github.com/go-openapi/jsonreference@v0.19.0 github.com/PuerkitoBio/purell@v1.1.0
github.com/go-openapi/jsonreference@v0.17.0 github.com/PuerkitoBio/purell@v1.1.0
```

Reply:
all the commands (including the initialization commands used to take master from current state to this PR) are now baked into `hack/update-vendor.sh`